x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/fuse/Fuse.java

```diff
@@ -95,4 +99,30 @@ public MapExpression options() {
     public boolean expressionsResolved() {
         return score.resolved() && discriminator.resolved() && groupings.stream().allMatch(Expression::resolved);
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        if (score.dataType() != DataType.DOUBLE) {
```

Comment:
I made a note in https://github.com/elastic/elasticsearch/issues/123389 that as a follow up, we might want to accept any numeric type for the score column.
I have not done this yet, since (1) the operator expects it to a double and (2) it would mean refactoring `FuseScoreEval`/`FuseScoreEvalExec` to be able to receive both an input score attribute and an output score attribute.